demos/common/cpp/models/src/hpe_model_associative_embedding.cpp

```diff
@@ -0,0 +1,207 @@
+        outputsNames.push_back(outputLayer.first);
+        const SizeVector& outputLayerDims = outputLayer.second->getTensorDesc().getDims();
+        if (outputLayerDims.size() != 4 && outputLayerDims.size() != 5)
+                throw std::runtime_error("output layers are expected to be 4-dimensional or 5-dimensional");
+        if (outputLayerDims[0] != 1 || outputLayerDims[1] != 17)
+                throw std::runtime_error("output layers are expected to have 1 batch size and 17 channels");
+    }
+}
+
+void HpeAssociativeEmbedding::reshape(CNNNetwork& cnnNetwork) {
+    ICNNNetwork::InputShapes inputShapes = cnnNetwork.getInputShapes();
+    SizeVector& InputLayerDims = inputShapes.begin()->second;
+    if (!targetSize) {
+        targetSize =  static_cast<int>(std::min(InputLayerDims[2], InputLayerDims[3]));
+    }
+    int inputHeight = static_cast<int>(std::round(targetSize * aspectRatio));
+    int inputWidth = targetSize;
+    if (aspectRatio >= 1.0) {
+        std::swap(inputHeight, inputWidth);
+    }
+    int height = static_cast<int>((inputHeight + stride - 1) / stride) * stride;
+    int width = static_cast<int>((inputWidth + stride - 1) / stride) * stride;
+    InputLayerDims[0] = 1;
+    InputLayerDims[2] = height;
+    InputLayerDims[3] = width;
+    inputLayerSize = cv::Size(InputLayerDims[3], InputLayerDims[2]);
+    cnnNetwork.reshape(inputShapes);
+}
+
+std::shared_ptr<InternalModelData> HpeAssociativeEmbedding::preprocess(const InputData& inputData, InferRequest::Ptr& request) {
+    auto& image = inputData.asRef<ImageInputData>().inputImage;
+    cv::Mat resizedImage;
+    float scale = std::min(inputLayerSize.height / static_cast<float>(image.rows),
+                           inputLayerSize.width / static_cast<float>(image.cols));
+    cv::resize(image, resizedImage, cv::Size(), scale, scale, cv::INTER_CUBIC);
+    int h = resizedImage.rows;
+    int w = resizedImage.cols;
+    cv::Mat paddedImage;
+    int bottom = inputLayerSize.height - resizedImage.rows;
+    int right = inputLayerSize.width - resizedImage.cols;
+    cv::copyMakeBorder(resizedImage, paddedImage, 0, bottom, 0, right,
+                       cv::BORDER_CONSTANT, meanPixel);
+    Blob::Ptr frameBlob = request->GetBlob(inputsNames[0]);
+    matU8ToBlob<uint8_t>(paddedImage, frameBlob);
+    return std::shared_ptr<InternalModelData>(new InternalScaleData(image.cols / static_cast<float>(w),
+                                                                    image.rows / static_cast<float>(h)));
+}
+
+std::unique_ptr<ResultBase> HpeAssociativeEmbedding::postprocess(InferenceResult& infResult) {
+    HumanPoseResult* result = new HumanPoseResult;
+    *static_cast<ResultBase*>(result) = static_cast<ResultBase&>(infResult);
+
+    auto aembdsMapped = infResult.outputsData[outputsNames[0]];
+    auto heatMapsMapped = infResult.outputsData[outputsNames[1]];
+    auto nmsHeatMapsMapped = infResult.outputsData[outputsNames[2]];
```

Comment:
Swap names for `nmsHeatMapsMapped` and `nmsHeats`. Result of `infResult.outputsData[outputsNames[2]]` isn't mapped here yet. The same is for other two.